Model/Observer/RecommendSettings.php

```diff
@@ -0,0 +1,216 @@
+<?php
+declare(strict_types=1);
+
+namespace Algolia\AlgoliaSearch\Model\Observer;
+
+use Algolia\AlgoliaSearch\Api\RecommendManagementInterface;
+use Algolia\AlgoliaSearch\Helper\ConfigHelper;
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\App\Config\Storage\WriterInterface;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Exception\LocalizedException;
+
+class RecommendSettings implements ObserverInterface
+{
+    /**
+     * @var string
+     */
+    private $productId = '';
+
+    /**
+     * @param ConfigHelper $configHelper
+     * @param WriterInterface $configWriter
+     * @param ProductRepositoryInterface $productRepository
+     * @param RecommendManagementInterface $recommendManagement
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     */
+    public function __construct(
+        private readonly ConfigHelper                 $configHelper,
+        private readonly WriterInterface              $configWriter,
+        private readonly ProductRepositoryInterface   $productRepository,
+        private readonly RecommendManagementInterface $recommendManagement,
+        private readonly SearchCriteriaBuilder        $searchCriteriaBuilder
+    ){}
+
+    /**
+     * @param Observer $observer
+     * @throws LocalizedException
+     */
+    public function execute(Observer $observer)
+    {
+        foreach ($observer->getData('changed_paths') as $changedPath) {
+            // Validate before enable FBT on PDP
+            if (
+                $changedPath == $this->configHelper::IS_RECOMMEND_FREQUENTLY_BOUGHT_TOGETHER_ENABLED
+                && $this->configHelper->isRecommendFrequentlyBroughtTogetherEnabled()
+            ) {
+                $this->validateFrequentlyBroughtTogether($changedPath);
+            }
+
+            // Validate before enable FBT on cart page
+            if (
+                $changedPath == $this->configHelper::IS_RECOMMEND_FREQUENTLY_BOUGHT_TOGETHER_ENABLED_ON_CART_PAGE
+                && $this->configHelper->isRecommendFrequentlyBroughtTogetherEnabledOnCartPage()
+            ) {
+                $this->validateFrequentlyBroughtTogether($changedPath);
```

Comment:
Unless I'm missing something I feel like we should be validating only once for each model. e.g. if the FBT model passed for PDP it should pass for cart no?

Reply:
I thought about it, however, I feel like that will make spaghetti code. To me this approach gives it better readability.

Reply:
Hmmm. Instead we have extra processing... I'm not sure nesting conditionals would make it less readable?

Reply:
Not sure which nesting approach are you referring to. Please provide an example

Reply:
Something like : 
```                
if (
    (
        $changedPath == ConfigHelper::IS_RECOMMEND_FREQUENTLY_BOUGHT_TOGETHER_ENABLED 
        && $this->configHelper->isRecommendFrequentlyBroughtTogetherEnabled()
    ) || (
        $changedPath == ConfigHelper::IS_RECOMMEND_FREQUENTLY_BOUGHT_TOGETHER_ENABLED_ON_CART_PAGE
        && $this->configHelper->isRecommendFrequentlyBroughtTogetherEnabledOnCartPage()
    ) 
) {
    $this->validateFrequentlyBroughtTogether($changedPath);
}
```
If I understand the discussion ?

Reply:
That works for me.

Reply:
Exactly what I was thinking @damcou Thank you!